Add startswith to Python strings (#1399)
Codecademy/docs: content/python/concepts/strings/terms/startswith/startswith.md

---
Title: '.startswith()'
Description: 'Checks whether or not a string starts with a given value.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Functions'
  - 'Methods'
  - 'Strings'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
  - 'paths/data-science'
---

> The `.startswith()` method checks a value against a given string and returns `True` if the string starts with that value. Otherwise, it returns `False`.

Reviewer: ```suggestion
The **`.startswith()`** method checks a value against a given string and returns `True` if the string starts with that value. Otherwise, it returns `False`.
```

## Syntax

```pseudo
string.startswith(value, start, end)
```

Given a string, the `.startswith()` method can be used in the following way:

> - The required `value` argument, which is checked if it exists at the beginning of the string. It is also case-sensitive.

Reviewer 2: ```suggestion
- The required `value` argument is checked for existence at the beginning of the `string`. It is also case-sensitive.
```

> 

Reviewer 2: ```suggestion
```

> - Optionally, the `value` can be tested for whether it is at the beginning of a particular substring within the string using the `start` and `end` index arguments.

Reviewer 2: ```suggestion
- Optionally, a `start` and `end` index argument can be declared. The `value` is then checked for existence at the beginning of the substring from the `string`'s `start` and `end` indices.
```

## Example

```py
example_str = "This is a string"

check_A = example_str.startswith("T")

> check_B = example_str.startswith("h")

Reviewer 2: ```suggestion
check_B = example_str.startswith("t")
```

Can we change this to show it's case-sensitivity?

> check_C = example_str.startswith("st", 10, 14)

Reviewer 2: I'd like to see this moved to the code byte because it's slightly ramped up to use the optional parameters.

print("A: ", check_A)
print("B: ", check_B)

> print("C: ", check_C)

Reviewer 2: This print statement and the output would need to be removed.

```

The output will look like this:

```shell
A: True
B: False
C: True
```

> From the above example, for `check_A` the output is `True` as the `.startswith()` function checks whether the `example_str` string starts with the character "T".

Reviewer 2: ```suggestion
In the example above, `check_A` is `True` since the `example_str` string starts with the character `"T"`.
```
